client/components/GAuth.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { GoogleLogin } from 'react-google-login'
+import GoogleButton from 'react-google-button'
+import { refreshTokenSetup } from './utils/refreshToken'
+import getConfig from "next/config";
+
+
+const { publicRuntimeConfig } = getConfig();
+
+function Login({ parentCallback }) {
```

Comment:
I guess it should be renamed to some Google related stuff (better if it's the same as the filename):
```suggestion
function GLogin({ parentCallback }) {
```

Also, `parentCallback` should be renamed to something more meaningful (you can use `onSuccess` as the GoogleLogin component)
```suggestion
function Login(props) {
// `props.onSuccess === parentCallback` now
```

Reply:
The "parentCallback" name was not randomly chosen, it is a JSX attribute passed in the GLogin component's tag (<Login/>), it is used together with the useCallback() function to make the asynchronous nature of the login synchronous, thus assuring that the profile object has been received from the Google API before accessing the application's Postgres database